src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespacearound/InputWhitespaceAroundAllowEmptyTypesAndNonEmptyClasses2.java

```diff
@@ -23,29 +23,31 @@
 import java.util.function.Function;
 import java.util.function.Supplier;
 
-public class InputWhitespaceAroundAllowEmptyTypesAndNonEmptyClasses2{ // violation
+    class EmptyAndNonEmptyClasses2{ int x; } // violation ''{' is not preceded with whitespace'
```

Comment:
move such new classes to the end of file, to keep top public class be top in file.
make indentation of EmptyAndNonEmptyClasses2 to start from beginning of file.
same for other changes line this.